starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,215 @@
+// First off, here are some defines and includes. I forget what exactly needs what
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+// We include time.h so we can seed the RNG for default "smart" movement with time()
+#include <time.h>
+
+
+// Now let's include some helper macros
+// You might reasonably want to keep some of these
+#include "macros.c"
+
+
+// Now for our data structures. These are intentionally left rather basic, so add whatever you want to them
+struct ship{
+	unsigned char x, y;
+	unsigned short halite;
+} ships[4096];
+
+struct player{
+	unsigned char dropoffs[64], dropoff_count, ship_count, ship_yard[2];
+	unsigned short ships[1024];
+	unsigned int halite;
+} players[4] = {0};
+
+struct map{
+	unsigned short halite;
+	unsigned short occupied; //(unsigned short)-1 means it's not occupied
+} map[64][64];
+
+
+// Now for some global variables to help us keep track of things
+unsigned char dropoffs[256][2], mhh, mhw, my_id, map_size[2], number_of_players;
+unsigned short max_turn, turn_number;
+
+FILE *log_file;
```

Comment:
Assign to `NULL`

Reply:
linter removed setting this to NULL. Not sure why. I imagine there's a reason to not set global variables to NULL